flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/BufferProviderInputChannelBuilder.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.consumer;
+
+import org.apache.flink.core.memory.MemorySegment;
+import org.apache.flink.core.memory.MemorySegmentFactory;
+import org.apache.flink.runtime.io.network.LocalConnectionManager;
+import org.apache.flink.runtime.io.network.buffer.Buffer;
+import org.apache.flink.runtime.io.network.buffer.FreeingBufferRecycler;
+import org.apache.flink.runtime.io.network.buffer.NetworkBuffer;
+import org.apache.flink.runtime.io.network.partition.InputChannelTestUtils;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionID;
+
+import javax.annotation.Nullable;
+
+import static org.apache.flink.util.Preconditions.checkState;
+
+/**
+ * Builder for special {@link RemoteInputChannel} that correspond to buffer request, allow users to
+ * set InputChannelId and release state.
+ */
+public class BufferProviderInputChannelBuilder {
+	private SingleInputGate inputGate = new SingleInputGateBuilder().build();
+	private InputChannelID id = new InputChannelID();
+	private int maxNumberOfBuffers = Integer.MAX_VALUE;
+	private int bufferSize = 32 * 1024;
+	private boolean isReleased = false;
+
+	public BufferProviderInputChannelBuilder setInputGate(SingleInputGate inputGate) {
+		this.inputGate = inputGate;
+		return this;
+	}
+
+	public BufferProviderInputChannelBuilder setId(InputChannelID id) {
+		this.id = id;
+		return this;
+	}
+
+	public BufferProviderInputChannelBuilder setMaxNumberOfBuffers(int maxNumberOfBuffers) {
+		this.maxNumberOfBuffers = maxNumberOfBuffers;
+		return this;
+	}
+
+	public BufferProviderInputChannelBuilder setBufferSize(int bufferSize) {
```

Comment:
We can remove it if not used atm

Reply:
Removed the builder, thus the method is also removed.